### ISSUE TITLE
[release/7.0.1xx] Prep.sh --bootstrap is failing in CI

### ISSUE THREAD

**Issue by MichaelSimons** [dotnet/source-build#3118]
Failing CI (Microsoft internal link) https://dev.azure.com/dnceng/internal/_build/results?buildId=2044286&view=logs&j=94fd4010-a547-56e5-f0a4-9ada8f56b43c&t=af288ef0-5c96-5b58-c317-135ff1c6f0fd

```
/tarball/.dotnet/sdk/7.0.100/MSBuild.dll -nologo --property:ArchiveDir=/tarball/packages/archive/ -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/tarball/.dotnet/sdk/7.0.100/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/tarball/.dotnet/sdk/7.0.100/dotnet.dll -maxcpucount -target:Restore -verbosity:m /bl:artifacts/prep/bootstrap.binlog /fileLoggerParameters:LogFile=artifacts/prep/bootstrap.log /tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj
  Determining projects to restore...
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102: Unable to find package Microsoft.NET.HostModel with version (= 7.0.0-rtm.22522.1)
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 2200 version(s) in dotnet6-transport [ Nearest version: 6.0.12-servicing.22561.8 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 1008 version(s) in dotnet7-transport [ Nearest version: 7.0.1-servicing.22552.6 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 750 version(s) in dotnet5-transport [ Nearest version: 6.0.0-alpha.1.20420.3 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 482 version(s) in dotnet5 [ Nearest version: 5.0.0-preview.2.20125.16 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 319 version(s) in general-testing [ Nearest version: 7.0.1-servicing.22471.5 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 22 version(s) in dotnet-public [ Nearest version: 5.0.0-preview.1.20120.5 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 1 version(s) in dotnet-eng [ Nearest version: 5.0.0-alpha.1.19618.1 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 1 version(s) in dotnet3.1-transport [ Nearest version: 5.0.8-servicing.21315.4 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in xamarin
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4-1
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4-2
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4-3
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-DotNet-msbuild-Trusted-18d5aef
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet-libraries
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet-tools
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet3-transport
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet6
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet7
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102: Unable to find package Microsoft.NETCore.TestHost with version (= 7.0.0-rtm.22522.1)
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 2069 version(s) in dotnet6 [ Nearest version: 6.0.2-mauipre.1.22102.15 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 1000 version(s) in dotnet7-transport [ Nearest version: 7.0.1-servicing.22552.6 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 817 version(s) in dotnet5 [ Nearest version: 6.0.0-alpha.1.20420.3 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 312 version(s) in general-testing [ Nearest version: 7.0.1-servicing.22471.5 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 247 version(s) in dotnet7 [ Nearest version: 7.0.0-alpha.1.21508.10 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 197 version(s) in dotnet5-transport [ Nearest version: 5.0.17-servicing.22260.7 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 134 version(s) in dotnet3.1-transport [ Nearest version: 5.0.8-servicing.21315.4 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 119 version(s) in dotnet6-transport [ Nearest version: 6.0.12-servicing.22561.8 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 50 version(s) in dotnet-public [ Nearest version: 6.0.2-mauipre.1.22102.15 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 15 version(s) in dotnet3-transport [ Nearest version: 3.0.3-servicing.20066.3 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in xamarin
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4-1
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4-2
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4-3
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-6b7d1f4
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in darc-pub-DotNet-msbuild-Trusted-18d5aef
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet-eng
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet-libraries
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 0 version(s) in dotnet-tools
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102: Unable to find package runtime.linux-x64.Microsoft.NETCore.TestHost with version (= 7.0.0-rtm.22522.1)
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 2069 version(s) in dotnet6 [ Nearest version: 6.0.2-mauipre.1.22102.15 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 1000 version(s) in dotnet7-transport [ Nearest version: 7.0.1-servicing.22552.6 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 817 version(s) in dotnet5 [ Nearest version: 6.0.0-alpha.1.20420.3 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 312 version(s) in general-testing [ Nearest version: 7.0.1-servicing.22471.5 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 247 version(s) in dotnet7 [ Nearest version: 7.0.0-alpha.1.21508.10 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 198 version(s) in dotnet5-transport [ Nearest version: 5.0.17-servicing.22260.7 ]
/tmp/tmp.1XaTrmhlr2/buildBootstrapPreviouslySB.csproj : error NU1102:   - Found 136 version(s) in dotnet3.1-transport [ Nearest version: 5.0.8-servicing.21315.4 ]
```

**Comment by MichaelSimons** [comment]
@crummel - Can you please take a look?  TIA

**Comment by tmds** [comment]
Will this be solved soon-ish?

**Comment by MichaelSimons** [comment]
> Will this be solved soon-ish?

A fix has been checked in internally that will be released with 7.0.1.  If this is need now for development purposes, we could mirror the fix publicly now.

**Comment by crummel** [comment]
@omajid @tmds This should be fixed for now with #15045.  I'm still working on a long-term fix here - the root of the problem is that installer expects some things to have stable versions and some to have unstable versions but we don't currently have a way to specify both.

**Comment by MichaelSimons** [comment]
Closing as a workaround was implemented.  Fixing the underlying issue is tracked with #3146
